docs/UA-service-API.md

```diff
@@ -0,0 +1,456 @@
+
+### Response
+
+Successful requests will produce a "200 OK" response with an empty JSON body:
+
+```json
+{
+}
+```
+
+## GET /v1/query
+
+Search for URLs, titles, and text content matching a query string.
+
+___Query parameters___
+
+* q - string query to search for.
+* limit - (optional) integer maximum number of results to return.
+* since - (optional) integer timestamp; if present, only return
+  results more recently visited than this.
+* snippetSize - (optional) integer, one of -5/-15/-40/-64 for tiny,
+  medium, large, and huge snippet extracts, respectively.  In the
+  future, we will clarify this -- right now, it's a direct translation
+  to SQLite's numeric range.
+
+### Request
+
+```sh
+curl -v \
+-X GET \
+-H "Content-Type: application/json" \
+"http://localhost:9090/v1/query?q=reddit&limit=10&since=13000000&snippetSize=-15
+```
+
+### Response
+
+Successful requests will produce a "200 OK" response with a `results`
+array in the JSON body:
+
+```json
+{
+  "results": [
+    {
+      "url": "https://reddit.com/",
+      "title": "reddit - the front page of the internet",
+      "lastVisited": 15000000,
+      "snippet": "HTML escaped string with <b>bold tags</b> wrapping highlighted words"
+    }
+  ]
+}
+```
+
+## PUT /v1/stars/:url
+
+Mark the given URL as starred.
+
+___URL parameters___
+
+* url - string URL to star.
+
+___Body parameters___
+
+* session - integer session ID to associate star action with.
+* title (optional) - string title of starred URL.
+
+### Request
+
+```sh
+curl -v \
+-X PUT \
+-H "Content-Type: application/json" \
+"http://localhost:9090/v1/stars/https%3A%2F%2Freddit.com%2F%26test%3D%202 \
+-d '{
+  "session": 101,
+  "title": "reddit - the front page of the internet"
+}'
+```
+
+### Response
+
+Successful requests will produce a "200 OK" response with an empty JSON body:
+
+```json
+{
+}
+```
+
+## DELETE /v1/stars/:url
+
+Mark the given URL as not starred.
+
+___URL parameters___
+
+* url - string URL to unstar.
+
+___Body parameters___
+
+* session - integer session ID to associate unstar action with.
+
+### Request
+
+```sh
+curl -v \
+-X DELETE \
+-H "Content-Type: application/json" \
+"http://localhost:9090/v1/stars/https%3A%2F%2Freddit.com%2F%26test%3D%202 \
+-d '{
+  "session": 101
+}'
+```
+
+### Response
+
+Successful requests will produce a "200 OK" response with an empty JSON body:
+
+```json
+{
+}
+```
+
+## GET /v1/stars
+
+Get the currently starred URLs.
+
+### Request
+
+```sh
+curl -v \
+-X GET \
+-H "Content-Type: application/json" \
+"http://localhost:9090/v1/stars
+```
+
+### Response
+
+Successful requests will produce a "200 OK" response with an array
+`stars` in the JSON body:
+
+```json
+{
+  "stars": [
+    "https://www.mozilla.org/en-US/firefox/new/",
+    "https://reddit.com/"
+  ]
+}
+```
+
+## GET /v1/recentStars
+
+Get recently starred URLs, ordered from most to least recent.
+
+___Query parameters___
+
+* limit - (optional) integer maximum number of starred URLs to return.
```

Comment:
Is it optional? https://github.com/mozilla/tofino/blob/master/app/services/user-agent-service/server.js#L191